core/commands/configure_command.rb

```diff
@@ -55,11 +55,24 @@ def execute
 
   private
 
+  def create_docker_file(password)
+    path = File.expand_path('~/.config/mdbci/docker.md')
+    return ERROR_RESULT if @configuration.check_config_dir(path, @ui) == ERROR_RESULT
+
+    file = File.new(path, 'w')
+    file.write(password)
+    file.close
+    file.path
+  end
+
   def check_dock_credentials(docker_credentials)
-    cmd = "docker login --username #{docker_credentials['username']}" \
-          " --password '#{docker_credentials['password']}' #{docker_credentials['ci-server']}"
+    path = create_docker_file(docker_credentials['password'])
+    return ERROR_RESULT if path == ERROR_RESULT
 
+    cmd = "cat #{path} | docker login --username #{docker_credentials['username']}" \
```

Comment:
Instead of using the file as a temporary file you should provide an option to the `run_command_and_log` that will not print the sensitive output to the stdout.